modules/network/src/main/java/org/apache/ignite/internal/network/DefaultMessagingService.java

```diff
@@ -287,10 +302,9 @@ private CompletableFuture<NetworkMessage> invoke0(ClusterNode recipient, Channel
 
         long correlationId = createCorrelationId();
 
-        CompletableFuture<NetworkMessage> responseFuture = new CompletableFuture<NetworkMessage>()
-                .orTimeout(timeout, TimeUnit.MILLISECONDS);
+        CompletableFuture<NetworkMessage> responseFuture = new CompletableFuture<>();
 
-        requestsMap.put(correlationId, responseFuture);
+        requestsMap.put(correlationId, new TimeoutObject(timeout > 0 ? coarseCurrentTimeMillis() + timeout : 0, responseFuture));
```

Comment:
It seems that `orTimeout(0, SECONDS)` will timeout immediately, but the `TimeoutObject` treats 0 as 'no timeout at all', so this changes the behavior. I think we need to establish what we want here first.

Reply:
I took this code from our repository. It was for a thin client.
Anyway, timeout 0 is something specific that will be used in a test.

Reply:
Thin client treats 0 as "no timeout" (which is documented in the config javadoc), this might be different in other places.

Reply:
That is exactly why I wrote it in this way (TimeoutWorker#body).